src/content.ts

```diff
@@ -0,0 +1,55 @@
+import { Bee } from '@ethersphere/bee-js'
+import client from 'prom-client'
+import type { ContentConfig } from './config'
+import { logger } from './logger'
+import { register } from './metrics'
+
+const contentReuploadCounter = new client.Counter({
+  name: 'content_reupload_counter',
+  help: 'How many pinned content items were uploaded',
+})
+register.registerMetric(contentReuploadCounter)
+
+export class ContentManager {
+  private interval?: ReturnType<typeof setInterval>
+  private isReuploading = false
+
+  public async refreshContentReupload(beeApi: Bee): Promise<void> {
+    const pins = await beeApi.getAllPins()
+
+    if (!pins.length) {
+      logger.info(`no pins found`)
+    } else {
```

Comment:
If you have a condition like this, that should prevent running for example the main body of function/method based on some condition like here, I prefer instead of this `if/else` construct that creates another nesting level (which you generally want to prevent as it makes the code harder to read), to simply exit the function instead.

```suggestion
    if (!pins.length) {
      logger.info(`no pins found`)
      return
    }
```

The suggestion needs also removing the `else` branch ;-)